handlers/cluster.go

```diff
@@ -0,0 +1,93 @@
+package handlers
+
+import (
+	"encoding/json"
+	"github.com/atlas-api-helper/resources/cluster"
+	"github.com/atlas-api-helper/util"
+	responseHandler "github.com/atlas-api-helper/util/Responsehandler"
+	"github.com/atlas-api-helper/util/constants"
+	"github.com/atlas-api-helper/util/logger"
+	"github.com/gorilla/mux"
+	"net/http"
+	"time"
+)
+
+func setupClusterLog() {
+	util.SetupLogger("atlas-api-helper.handlers.cluster")
+}
+
+// GetCluster handles GET requests to return the state of the cluster
+func GetCluster(w http.ResponseWriter, r *http.Request) {
+	setupClusterLog()
+	vars := mux.Vars(r)
+	projectId := vars[constants.ProjectIdPathParam]
```

Comment:
func GetCluster(w http.ResponseWriter, r *http.Request) {
	setupClusterLog()

       // fetch parameters from request
	vars := mux.Vars(r)
	projectId := vars[constants.ProjectIdPathParam]
	Name := vars[constants.ClusterNamePathParam]
	publicKey := r.URL.Query().Get(constants.PublicKeyQueryParam)
	privateKey := r.URL.Query().Get(constants.PrivateKeyQueryParam)
	model := cluster.InputModel{ProjectId: &projectId, ClusterName: &Name, PrivateKey: &privateKey, PublicKey: &publicKey}

	_, _ = logger.Debugf("Get clusters request  starts: %s", model.String())
	startTime := time.Now()
       // API Request
	response := cluster.Read(&model)
	elapsedTime := time.Since(startTime)

	_,_,_=logger.Debugf("Get all Clusters REST API  response:%+v and execution time:%s", response.String(), elapsedTime.String())

	responseHandler.Write(response, w, constants.ClusterHandler)
}